packages/workflow/src/activities/executeInteraction.ts

```diff
@@ -187,14 +187,22 @@ export async function executeInteraction(payload: DSLActivityExecutionPayload<Ex
         });
 
     } catch (error: any) {
-        log.error("Failed to execute interaction", { error });
         if (error.message.includes("Failed to validate merged prompt schema")) {
+            log.error("Failed to validate merged prompt schema", { 
+                error, 
+                "@metadata": { "error.code": "validation_error" }
```

Comment:
Here you are using the logger from Temporal Activity (`import { log } from "@temporalio/activity";`) which wraps all the attributes under the field `metadata` in Temporal. So I think you don't need to prefix by "metadata".

```suggestion
                "error.code": "validation_error"
```

The character `@` stands for attributes, which is a specific character used by Datadog's Logs for querying attributes. But it does not need to be present in the logs themselves.